clientconn.go

```diff
@@ -119,6 +132,7 @@ type ClientConn struct {
 
 func (cc *ClientConn) resetTransport(closeTransport bool) error {
 	var retries int
+	totime := time.Now().Add(cc.dopts.timeout)
```

Comment:
Yes, being explicit with handling the `cc.dopts.timeout > 0` case seems wiser. I'd do `start := time.Now()` at the start of the function, and then do something like
```go
if d := cc.dopts.timeout; d > 0 && time.Since(start) > d {
  // oops, gone past timeout
}
```
when it's needed. Then the use of `cc.dopts.timeout` is crystal clear and you don't have to trace its implicit effects through other variables.